spec/ruby/core/module/name_spec.rb

```diff
@@ -6,6 +6,40 @@
     Module.new.name.should be_nil
   end
 
+  ruby_version_is "3.3" do
+    it "can assign a temporary name" do
+      m = Module.new
+      m.name.should be_nil
+
+      m.set_temporary_name("Foo")
+      m.name.should == "Foo"
+
+      m.set_temporary_name(nil)
+      m.name.should be_nil
+    end
+
+    it "can't assign empty string as name" do
+      m = Module.new
+      -> { m.set_temporary_name("") }.should raise_error(ArgumentError, "empty class/module name")
+    end
+
+    it "can't assign name to permanent module" do
+      -> { Object.set_temporary_name("Foo") }.should raise_error(RuntimeError, "can't change permanent name")
+    end
+
+    it "can assign a temporary name to a nested module" do
+      m = Module.new
+      module m::N; end
+      m::N.name.should =~ /\A#<Module:0x\h+>::N\z/
+
+      m::N.set_temporary_name("Foo")
+      m::N.name.should == "Foo"
```

Comment:
This is also very strange, that module has a lexical parent (m), but its Module#name does not reflect it.
Again it's a very confusing "lie" like above.
If it's `foo` or `<#tmp name:Foo>` or so then there is no confusion with a constant path.